src/saltext/vmware/states/vmc_dhcp_profiles.py

```diff
@@ -0,0 +1,390 @@
+        If set to False, the certificate validation is skipped.
+
+    cert
+        (Optional) Path to the SSL client certificate file to connect to VMC Cloud Console.
+        The certificate can be retrieved from browser.
+
+    server_addresses
+        when type is relay, this field indicates DHCP relay addresses(DHCP server IP addresses
+        for DHCP relay configuration).
+        Both IPv4 and IPv6 addresses are supported.
+
+        when type is server, this field indicates DHCP server address in CIDR format.
+        Both IPv4 and IPv6 address families are supported.
+        Prefix length should be less than or equal to 30 for IPv4 address family
+        and less than or equal to 126 for IPv6.
+        When not specified, IPv4 value is auto-assigned to 100.96.0.1/30.
+
+        Note: This field is optional only when type is server and
+        is mandatory when type is relay.
+
+    tags
+        (Optional) Opaque identifiers meaningful to the user.
+
+        .. code::
+
+            tags:
+              - tag: <tag-key-1>
+                scope: <tag-value-1>
+              - tag: <tag-key-2>
+                scope: <tag-value-2>
+
+    lease_time
+        (Optional) IP address lease time in seconds. Minimum value is 60.
+        Maximum value is 4294967295. Default value is 86400
+        Note: This field is applicable only when type is server
+
+    display_name
+        Identifier to use when displaying entity in logs or GUI.
+
+    Example values:
+
+        .. code::
+
+            for dhcp-server-profiles:
+
+            server_addresses:
+              - 10.22.12.2/23
+            tags:
+              - tag: tag1
+                scope: scope1
+            lease_time: 86400
+
+            for dhcp-relay-profiles:
+
+            server_addresses:
+              - 10.1.1.1
+            tags:
+              - tag: tag1
+                scope: scope1
+
+    """
+
+    input_dict = {
+        "server_addresses": server_addresses,
+        "tags": tags,
+        "lease_time": lease_time,
+        "display_name": display_name,
+    }
+
+    if type == vmc_constants.RELAY and lease_time:
+        return vmc_state._create_state_response(
+            name=name, comment="lease_time is not applicable for DHCP Relay Profile", result=False
+        )
+
+    if type == vmc_constants.RELAY:
+        input_dict.pop("lease_time")
+
+    input_dict = {k: v for k, v in input_dict.items() if v != vmc_constants.VMC_NONE}
+
+    dhcp_profile = __salt__["vmc_dhcp_profiles.get_by_id"](
+        hostname=hostname,
+        refresh_key=refresh_key,
+        authorization_host=authorization_host,
+        org_id=org_id,
+        sddc_id=sddc_id,
+        type=type,
+        dhcp_profile_id=dhcp_profile_id,
+        verify_ssl=verify_ssl,
+        cert=cert,
+    )
+
+    if "error" in dhcp_profile:
+        if "could not be found" in dhcp_profile["error"]:
+            dhcp_profile = None
```

Comment:
**suggestion** there is no test case for this (i.e. add `assert False` here and the tests do not fail)

Reply:
added testcase